### PR TITLE
CHE-11954: Add che.registerTaskRunner Theia Plugin API

### PR DESCRIPTION
**Pull request by vinokurig**
**Do not merge** until Theia 0.3.19 is released
Needs to change the version of `@theia/core` and `@theia/plugin-ext` to 0.3.19 after release

### CODE REVIEW

File: extensions/eclipse-che-theia-plugin-ext/package.json [hunk 1]
```diff
@@ -15,8 +15,8 @@
     "dependencies": {
         "@eclipse-che/plugin": "latest",
         "@eclipse-che/workspace-client": "^0.0.1-1546509769",
-        "@theia/core": "0.3.18",
-        "@theia/plugin-ext": "0.3.18"
+        "@theia/core": "0.4.0-next.b0064648",
```
Comment (vitaliy-guliy) [inline extensions/eclipse-che-theia-plugin-ext/package.json]: Probably use 0.3.19 here?

File: extensions/eclipse-che-theia-plugin-ext/src/plugin/che-api.ts [hunk 2]
```diff
@@ -78,10 +80,20 @@ export function createAPIFactory(rpc: RPCProtocol): CheApiFactory {
...
+            fireTaskExited(id: number): Promise<void> {
+                return cheTaskImpl.fireTaskExited(id);
+            }
+        };
+
         return <typeof che>{
             workspace,
             factory,
-            variables
+            variables: variables,
```
Comment (vitaliy-guliy) [inline extensions/eclipse-che-theia-plugin-ext/src/plugin/che-api.ts]: Use just `variables`

File: extensions/eclipse-che-theia-plugin-ext/package.json (hunk 1 above)
Comment (benoitf) [reply extensions/eclipse-che-theia-plugin-ext/package.json]: yes, need to use tagged version there please (0.3.19 should come next week)

Comment (vinokurig) [reply extensions/eclipse-che-theia-plugin-ext/package.json]: I'll change it after the release

Comment (benoitf) [reply extensions/eclipse-che-theia-plugin-ext/package.json]: thanks

File: extensions/eclipse-che-theia-plugin-ext/src/plugin/che-api.ts (hunk 2 above)
Comment (vinokurig) [reply extensions/eclipse-che-theia-plugin-ext/src/plugin/che-api.ts]: fixed

**Pull request merged**